app/upgrades/v7/upgrades.go

```diff
@@ -0,0 +1,322 @@
+		auctionInitGenesis(ctx, auctionKeeper)
+
+		ctx.Logger().Info("v7 upgrade: initializing pubsub genesis state")
+		pubsubInitGenesis(ctx, pubsubKeeper)
+
+		ctx.Logger().Info("v7 upgrade: running migrations and exiting handler")
+		return mm.RunMigrations(ctx, configurator, fromVM)
+	}
+}
+
+// Initialize the auction module with prices for some stablecoins and SOMM.
+func auctionInitGenesis(ctx sdk.Context, auctionKeeper auctionkeeper.Keeper) {
+	genesisState := auctiontypes.DefaultGenesisState()
+
+	usomm52WeekLow := sdk.MustNewDecFromStr("0.079151")
+	oneDollar := sdk.MustNewDecFromStr("1.0")
+	// TODO(bolten): update LastUpdatedBlock to the upgrade height when finalized
+	usommPrice := auctiontypes.TokenPrice{
+		Denom:            "usomm",
+		UsdPrice:         usomm52WeekLow,
+		Exponent:         6,
+		LastUpdatedBlock: 1,
+	}
+
+	// setting stables to 1 dollar
+	usdcPrice := auctiontypes.TokenPrice{
+		Denom:            "gravity0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+		UsdPrice:         oneDollar,
+		Exponent:         6,
+		LastUpdatedBlock: 1,
+	}
+
+	usdtPrice := auctiontypes.TokenPrice{
+		Denom:            "gravity0xdAC17F958D2ee523a2206206994597C13D831ec7",
+		UsdPrice:         oneDollar,
+		Exponent:         6,
+		LastUpdatedBlock: 1,
+	}
+
+	daiPrice := auctiontypes.TokenPrice{
+		Denom:            "gravity0x6B175474E89094C44Da98b954EedeAC495271d0F",
+		UsdPrice:         oneDollar,
+		Exponent:         18,
+		LastUpdatedBlock: 1,
+	}
+
+	genesisState.TokenPrices = []*auctiontypes.TokenPrice{&usommPrice, &usdcPrice, &usdtPrice, &daiPrice}
+
+	auctionkeeper.InitGenesis(ctx, auctionKeeper, genesisState)
+}
+
+// Set up the initial pubsub state to mirror what is currently used in practice already, with 7seas as the
+// first publisher using its existing CA certificate, its default subscriptions as the already launched cellars,
+// and the subscribers as reflected in the steward-registry repo.
+func pubsubInitGenesis(ctx sdk.Context, pubsubKeeper pubsubkeeper.Keeper) {
+	genesisState := pubsubtypes.DefaultGenesisState()
+
+	// Initialize the 7seas publisher.
+	publisher := pubsubtypes.Publisher{
+		Address: "somm14zsm5frvjuqxk3f9377altc6xq368dglhmkxmp",
+		Domain:  SevenSeasDomain,
+		CaCert:  SevenSeasPublisherCA,
+	}
+	publishers := []*pubsubtypes.Publisher{&publisher}
+
+	// TODO(bolten): update the cellars list with the current total cellars
+	cellars := []string{
+		"0x7bAD5DF5E11151Dc5Ee1a648800057C5c934c0d5", // Aave V2
+		"0x6b7f87279982d919bbf85182ddeab179b366d8f2", // ETH-BTC Trend
+		"0x6e2dac3b9e9adc0cbbae2d0b9fd81952a8d33872", // ETH-BTC Momentum
+		"0x3F07A84eCdf494310D397d24c1C78B041D2fa622", // Steady ETH
+		"0x4986fD36b6b16f49b43282Ee2e24C5cF90ed166d", // Steady BTC
+		"0x05641a27C82799AaF22b436F20A3110410f29652", // Steady MATIC
+		"0x6f069f711281618467dae7873541ecc082761b33", // Steady UNI
+		"0x97e6E0a40a3D02F12d1cEC30ebfbAE04e37C119E", // Real Yield USD
+	}
```

Comment:
also need to update this with all more recent cellars

Reply:
Yes, will have the handler fully up-to-date in an upcoming commit.